3-platform/shopping-cart/app/src/main/java/com/_6core/platform/shopping/cart/app/validator/impl/CartNotEmpty.java

```diff
@@ -0,0 +1,15 @@
+package com._6core.platform.shopping.cart.app.validator.impl;
+
+import com._6core.platform.shopping.cart.app.validator.ValidationHandler;
+import com._6core.platform.shopping.cart.domain.ShoppingCartDomain;
+
+/** Example of possible validation rule implementation. Can be changed or superseded. */
+public class CartNotEmpty extends ValidationHandler<ShoppingCartDomain> {
```

Comment:
Why concrete class? Lets depend on abstraction
```suggestion
public class CartNotEmpty extends ValidationHandler<ShoppingCartV01> {
```